src/java/org/apache/cassandra/dht/RangeStreamer.java

```diff
@@ -353,9 +361,20 @@ public void addRanges(String keyspaceName, ReplicaCollection<?> replicas)
      */
     private boolean useStrictSourcesForRanges(AbstractReplicationStrategy strat)
     {
-        return useStrictConsistency
-                && tokens != null
-                && metadata.getSizeOfAllEndpoints() != strat.getReplicationFactor().allReplicas;
+        int nodes = 0;
+
+        if (strat instanceof NetworkTopologyStrategy)
```

Comment:
We could add a `getReplicationFactor(dc)` to `SimpleStrategy` to collapse this further thx to inheritance. But that would mean `SimpleStrategy.getReplicationFactor(dc)` would have to compare the dc strings to match. That would go against `AbstractReplicationStrategy.getReplicationFactor()` notes stating this should be FAST. The extra string comparison is the problem so I opted for this solution. Happy to discuss.

Reply:
I like that but I would spare creating a new function.